contracts/temporary/FlowFreeze.cdc

```diff
@@ -0,0 +1,47 @@
+/*
+
+    FlowFreeze
+
+    This contract gives an admin the ability to freeze/unfreeze accounts
+    by adding them to a list stored in the contract.
+
+    If the account is on the frozen list, it will not be able to send transactions
+    and any transaction that accesses its account state will fail.
+
+    The execution envorionment will automatically query this contract as part of every transaction.
+
+    This is a TEMPORARY CONTRACT that will be in place until the proper protections
+    are in place with fees and the network is more decentralized
+
+ */
+
+
+pub contract FlowFreeze {
+
+    access(account) var freezeList: {Address: Bool}
+
+    pub let AdminStoragePath: StoragePath
+
+    pub resource Admin {
+
+        pub fun freezeAccount(_ address: Address) {
+            FlowFreeze.freezeList[address] = true
+        }
+
+        pub fun unfreezeAccount(_ address: Address) {
+            FlowFreeze.freezeList.remove(key: address)
+        }
+    }
+
+    pub fun getFreezeList(): {Address: Bool} {
+        return FlowFreeze.freezeList
```

Comment:
Copy-on-write makes this efficient, I need to train myself out of being wary of returning composite types. :-)

Reply:
Just to prevent misunderstandings: There's no copy-on-write optimization yet, but that shouldn't matter when writing contracts

Reply:
@turbolent Ah OK, thank you! Returning this map and then searching it in a script (as below) should still be almost as efficient as searching it in a function in the contract, though?